packages/react-redux/src/slices/cartoSlice.js

```diff
@@ -93,6 +94,30 @@ export const createCartoSlice = (initialState) => {
       setViewPort: (state) => {
         state.viewport = new WebMercatorViewport(state.viewState).getBounds();
       },
+      addSpatialFilter: (state, action) => {
+        const { sourceId, geometry } = action.payload;
+        if (sourceId) {
+          const source = state.dataSources[sourceId];
+
+          if (source) {
+            source.spatialFilter = geometry;
+          }
+        } else {
+          state.spatialFilter = geometry;
+        }
+      },
+      removeSpatialFilter: (state, action) => {
+        const sourceId = action.payload;
+        if (sourceId) {
+          const source = state.dataSources[sourceId];
+
+          if (source) {
+            source.spatialFilter = null;
+          }
+        } else {
+          state.spatialFilter = null;
+        }
+      },
       addFilter: (state, action) => {
```

Comment:
I think that `clearFilters` action should also remove the spatial filter for the datasource. Do you agree @VictorVelarde and @Clebal ?

Reply:
I've not been summoned but I'd say it shouldn't. The `clearFilters` action should remove the filters added by the `addFilter` action. As we have decided to not treat the spatial filter like the other filters, we shouldn't remove it using this action.

Reply:
Agree with @borja-munoz